cpp/ql/src/experimental/library/ArrayLengthAnalysis.qll

```diff
@@ -0,0 +1,274 @@
+/**
```

Comment:
The directory structure under `experimental` should mirror the structure of `src`, so I'd like to move this library to a directory that mirrors it future location outside `experimental`. I think it could belong in `semmle/code/cpp/rangeanalysis`. What do you think, @rdmarsh2?

Reply:
That sounds right.